eclipse/src/saros/Saros.java

```diff
@@ -260,29 +269,31 @@ private void stopLifeCycle() {
   }
 
   private void setupLoggers() {
-    /*
-     * HACK this is not the way OSGi works but it currently fulfill its
-     * purpose
-     */
-    final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
 
     final boolean isDebugMode = Boolean.getBoolean("saros.debug") || isDebugging(); // $NON-NLS-1$
 
-    final String log4jPropertyFile =
-        isDebugMode
-            ? "saros_debug.log4j.properties" //$NON-NLS-1$
-            : "saros_release.log4j.properties"; //$NON-NLS-1$
-
     try {
-      // change the context class loader so Log4J will find the appenders
-      Thread.currentThread().setContextClassLoader(Saros.class.getClassLoader());
+      final File pluginLogDir = new File(getStateLocation().toFile(), "log");
+      final Level logLevel = isDebugMode ? Level.ALL : Level.WARN;
+
+      URL configFileUrl = getBundle().getResource(LOG4J2_CONFIG_FILENAME);
```

Comment:
Just for my understanding, is the returned URL a file URL or a classLoader URL ?

After doing some googlestuff

you could replace line 280 of the old logic with

`LoggerContext context = Configurator.initialize(null, ConfigurationSource.fromResource(LOG4J2_CONFIG_FILENAME, Saros.class.getClassLoder())`

Reply:
Works with changing the default context selector :+1: